### ISSUE TITLE
String.format does not support thousands grouping

### ISSUE THREAD

**Issue by matthughes** [scala-js/scala-js#1619]
The following code produces different results whether run in browser vs jvm.  ScalaJS does not appear to support the thousands separator.

```
object ScalaJSExample extends js.JSApp{
  def main(): Unit = {
    println(f"${1000}%,d")    // prints 1000, not 1,000
    println(f"${1000F}%,.2f") // prints 1000.00, not 1,000.00
  }
}
```

**Comment by sjrd** [comment]
Indeed, it does not. PR welcome :)

**Comment by matthughes** [comment]
So... one could just hack in the *english* thousands separator of a comma but that doesn't really respect the API as grouping separator is defined by the locale.  By the logic, using . as the decimal point is also invalid as in German it should be a comma.

http://docs.oracle.com/javase/7/docs/api/java/util/Formatter.html
  	The result will include locale-specific grouping separators

However, I'm not sure how to get the current Locale as seen in the browser and can't see any other place in ScalaJS where we deal with locales.

There is this method on Number: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Number/toLocaleString but this does not work in even the latest Safari.

Having said all that, I'm beginning to see why this hasn't been implemented yet.

**Comment by matthughes** [comment]
Given the decimal comment, is ScalaJS just declaring itself to use the 'en' locale?

**Comment by matthughes** [comment]
Would you accept a PR that at least added the toLocaleString method on NumberOps?  It is at least defined by every browser.  Only Safari doesn't support the locales and all the options.

```
trait JSNumberOps extends Any {

  def toLocaleString(locales: Array[String]): String = native
  def toLocaleString(locales: Array[String], options: Dictionary[Any]): String = native
}
```

**Comment by sjrd** [comment]
> Given the decimal comment, is ScalaJS just declaring itself to use the 'en' locale?

No, not really. But we had to implement at least the decimal separator to be able to run the Scala test suite with Scala.js and get reasonable results.

Defining `toLocaleString` with `locales` is OK. It needs the `ecmascript6` badge in its ScalaDoc, though. An example can be found here: https://github.com/scala-js/scala-js/blob/master/library/src/main/scala/scala/scalajs/js/typedarray/Int32Array.scala#L5

But our implementation of the JDK libs should not rely on any non-es5 stuff. Otherwise it links, and then people think it works, but in fact it crashes in some browsers. If necessary, es6-dependent stuff can go in javalibex instead, which is a dependency that must be added manually by developers, thereby making them conscious they're using some features that might not be supported everywhere.

**Comment by sjrd** [comment]
Fixed in ab1d39777e6e9a9147265dfecbb76f33597ee035.
